### PR TITLE
CHEF-4554 : GET /{sku}/relatedProducts API

### PR DESCRIPTION
**Pull request by shanmugapriya-tr**
This PR contains changes for the new GET API  /{sku}/relatedProducts . It queries the DynamoDB and gets the related products for a given SKU

## Related Issue
https://chefio.atlassian.net/browse/CHEF-4554

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

- Response from the API ( Success ) 
Screenshot 2023-08-24 at 4 06 10 PM
- Response form the api if SKU is not present in DB 
Screenshot 2023-08-24 at 4 07 05 PM
Demo Video : 
https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/EemHW1PDUyFGl3rzwvkh8YMBE1HJYhFSwWTeWFlcFnzkcQ?e=YeYQVj

### CODE REVIEW

**Review by vipin230 — COMMENTED** [review]
Please rebase branch once as there are some changes in db operations PR. also remove cover.out file from PR

**Review by vviveksharma — APPROVED** [review]

**Review by gangwarshivendra — APPROVED** [review]
Looks good to me.

**Review by vipin230 — APPROVED** [review]

File: services/main.go [hunk 1]
```diff
@@ -353,3 +354,44 @@ func (server *ApiService) productDownloadHandler(c *fiber.Ctx) error {
...
 }
+
+// @description Get related products for a given SKU
+// @description The `ACCEPT` HTTP header with a value of `application/json` must be provided in the request for a JSON response to be returned
+// @Param       sku    	   query string true  "sku"
+// @Param       license_id query string false "License ID"
+// @Success     200
+// @Failure     400 {object} services.ErrorResponse
+// @Failure     403 {object} services.ErrorResponse
+// @Router      /{sku}/relatedProducts [get]
```
Comment (vipin230) [inline services/main.go:365]:
```suggestion
// @Router      /relatedProducts [get]
```

File: dboperations/dboperations.go [hunk 2]
```diff
@@ -121,22 +122,26 @@ func (dbo *DbOperationsService) GetVersionLatest(partitionValue string) (string,
 func (dbo *DbOperationsService) GetRelatedProducts(partitionValue string) (*models.RelatedProducts, error) {
 	res, err := dbo.fetchDataValues(partitionValue, dbo.skuTableName, SKU_PARTITION_KEY)
 	if err != nil {
-		log.Printf("error in fetching the database values: %v", err)
+		log.Errorf("error in fetching the database values: %v", err)
 		return nil, err
 	}
+
+	lenght := len(res.Items)
```
Comment (vipin230) [inline dboperations/dboperations.go:129]:
```suggestion
	length := len(res.Items)
```

Comment (vviveksharma) [reply dboperations/dboperations.go]: Updated the spelling

File: services/main.go (hunk 1 above)
Comment (vviveksharma) [reply services/main.go]: changed the comment for usage

File: services/main.go [hunk 3]
```diff
@@ -353,3 +354,44 @@ func (server *ApiService) productDownloadHandler(c *fiber.Ctx) error {
...
+		server.Log.Error("Validation of related products API for "+params.SKU+"failed", err.Error())
+		return err
+	}
+
+	relatedProducts, err := server.DatabaseService.GetRelatedProducts(params.SKU)
+
+	if err != nil {
+		request := clients.Request{}
+		server.Log.Error("Error while fetching related products for "+params.SKU, err.Error())
+		return server.SendError(c, request.Failure(500, "Unable to retrieve related products for "+params.SKU))
```
[the remark this answers is not in the document]
Comment (vviveksharma) [reply services/main.go]: Done

File: services/main.go [hunk 4]
```diff
@@ -353,3 +354,44 @@ func (server *ApiService) productDownloadHandler(c *fiber.Ctx) error {
...
+	}
+
+	if len(relatedProducts.Products) == 0 {
+		request := clients.Request{}
+		server.Log.Error("No related products found for " + params.SKU)
+		return server.SendError(c, request.Failure(400, "No related products found for SKU"))
```
[the remark this answers is not in the document]
Comment (vviveksharma) [reply services/main.go]: Done

**Review by vipin230 — APPROVED** [review]

**Review by vviveksharma — APPROVED** [review]

**Pull request merged**